tests/base/dataset_test.py

```diff
@@ -747,26 +746,60 @@ def test_train_test_split_changed(iris):
 
 def test_inferred_label_type_cat(diabetes_df):
     # Arrange
-    label = diabetes_df['target']
+    label = diabetes_df['target'].rename('actual')
     data = diabetes_df.drop('target', axis=1)
-    dataset = Dataset(data, label, label_name='actual')
+    dataset = Dataset(data, label)
     # Assert
     assert_that(dataset.label_type, is_('regression_label'))
 
 
 def test_inferred_label_type_reg(iris):
     # Arrange
-    label = iris['target']
+    label = iris['target'].rename('actual')
     data = iris.drop('target', axis=1)
-    dataset = Dataset(data, label, label_name='actual')
+    dataset = Dataset(data, label)
     # Assert
     assert_that(dataset.label_type, is_('classification_label'))
 
 
 def test_set_label_type(iris):
     # Arrange
-    label = iris['target']
+    label = iris['target'].rename('actual')
     data = iris.drop('target', axis=1)
-    dataset = Dataset(data, label, label_name='actual', label_type='regression_label')
+    dataset = Dataset(data, label, label_type='regression_label')
     # Assert
     assert_that(dataset.label_type, is_('regression_label'))
+
```

Comment:
Tests for invalid label types?
e.g. DataFrame with multiple columns, numpy array with incorrect dimensions, string that doesn't exist in dataframe, etc.

Reply:
lol TBH I didn't know we had a "label type" argument, cool
I actually meant the "improper shapes of np arrays and/or dataframes"